service/history/workflow/mutable_state_impl.go

```diff
@@ -4966,7 +4940,7 @@ func (ms *MutableStateImpl) updateActivityInfoForRetries(
 	})
 }
 
-func (ms *MutableStateImpl) UpdateActivityWithCallback(activityId string, updateCallback UpdateActivityCallback) error {
+func (ms *MutableStateImpl) UpdateActivity(activityId string, updateCallback UpdateActivityCallback) error {
```

Comment:
I (and future readers) would also appreciate the comment on why `updater` is needed here.

Reply:
I mean update operation requires callback at all.